### ISSUE TITLE
a single devcontainer cannot be connected to more than once

### ISSUE THREAD

**Issue by evbo** [microsoft/vscode#189937]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 
- OS Version: 

Steps to Reproduce:

1. docker compose that mounts directory above multiple projects
2. connect to the container and open each path so its in recent history. This works fine from a single VS Code Window.
3. Where the issue begins is if you open another vs code window and try to connect to one of the sub folders, e.g. "project_a" in my example below

results in error:
> Error running command remote-containers.openFolderInContainerInCurrentWindow: A system error occurred (ENOENT: no such file or directory, scandir '/workspace/project_b'). This is likely caused by the extension that contributes remote-containers.openFolderInContainerInCurrentWindow

![drone1](https://github.com/microsoft/vscode/assets/6313004/857675b3-bf55-4f40-8d22-9148049fa79e)

example compose file:
```yaml
version: "3.8"

services:
  vscode:
    image: "ubuntu"
    build:
      context: "../.."
      dockerfile: "./project_a/.devcontainer/Dockerfile"
    volumes:
      # project_a and project_b are folders inside this directory...
      - ../..:/workspace
    # docker run -i / interactive
    stdin_open: true
    # docker run -t / terminal
    tty: true

```

In summary, all paths exist and can be opened without any issues from a single VS Code window. But once you open another VS Code window you then cannot connect to that container at all.

This is useful for projects that have identical environments so that you can connect to them on a single container rather than maintaining multiple containers

**Comment by evbo** [comment]
This problem does not exist in version `v0.304.0`

I am switching temporarily to that version until this is fixed. With that version, you can open as many subfolders into your workspace in as many vs code windows as you want.

**Comment by chrmarti** [comment]
This works for me in the latest version. Using:

devcontainer.json:
```json
{
	"dockerComposeFile": "docker-compose.yml",
	"service": "app",
	"workspaceFolder": "/workspaces"
}
```

docker-compose.yml:
```yaml
version: '3.8'

services:
  app:
    image: mcr.microsoft.com/devcontainers/javascript-node:1-20-bullseye
    volumes:
      - ../..:/workspaces:cached
    command: sleep infinity
```
